### DISCUSSION TITLE
Installing a package directly from Github and importing it in a script

### DISCUSSION THREAD

**Discussion by holwech** [python-poetry/poetry#3603] [Q&A] [answered]
Hi everyone,

I'm trying to install a package directly from Github using poetry, but I'm struggling to get it to work the way I hope it can.

As a demo, [I've followed the docs](https://python-poetry.org/docs/cli/#add) and attempted to install `pendulum` by running the following command `poetry add git+https://github.com/sdispater/pendulum.git` which successfully installs the package to my environment. I then want to `import pendulum`, but this does not seem to work as I just get the error `No module named 'pendulum'`. 

Is it even possible to install a package in this way? If so, what do I have to do to be able to use the installed package in my code?

I'm planning on setting up my own repo that will work as an installable package, so it would be useful to know what the requirements are for a package to be installable.

Thanks!

**Comment by holwech** [comment, marked as answer]
I see now that I had activated the wrong environment, it seems to be as simple as running the poetry add command after all ;)

**Reply by recepdaban** [reply]
Hi, i could not get your answer. I have same problem with my github sample python project. i could add this project with poetry add command but could not import. When i try to run it says no module name. Should i switch to some environment ?
